src/cp/constraints/min_staffing.py

```diff
@@ -42,7 +41,7 @@ def create(self, model: CpModel, variables: dict[str, Variable]):
                 eligible_employees = self._get_eligible_employees(required_level)
 
                 for shift in self._shifts:
-                    if shift._id in self.EXCLUDED_SHIFTS:
+                    if shift._id in Shift.EXCLUSIVE_SHIFTS:
```

Comment:
Add a property to the `Shift` objects that decides whether a shift is exclusive or not. Not a list of shifts. Ids and indexes can change